src/cache/s3.rs

```diff
@@ -40,25 +33,52 @@ impl S3Cache {
         endpoint: Option<&str>,
         use_ssl: Option<bool>,
     ) -> Result<S3Cache> {
+        let mut builder = s3::Builder::default();
+        builder.bucket(bucket);
+        if let Some(region) = region {
+            builder.region(region);
+        }
+        builder.root(key_prefix);
+        if no_credentials {
+            builder.disable_credential_loader();
+        }
+        if let Some(endpoint) = endpoint {
+            let endpoint_uri: http::Uri = endpoint
+                .try_into()
+                .map_err(|err| anyhow!("input endpoint {endpoint} is invalid: {:?}", err))?;
+            let mut parts = endpoint_uri.into_parts();
+            match use_ssl {
```

Comment:
could you please keep this as in function (last the previous version)

Reply:
Added back.